sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientTest.java

```diff
@@ -443,6 +448,175 @@ void deadLetterWithDescription() {
+    }
+
+    /**
+     * Verifies that error source is populated when any error happened while renewing lock.
+     */
+    @Test
+    void errorSourceOnSessionLock() {
+        // Arrange
+        when(managementNode.renewSessionLock(SESSION_ID, null)).thenReturn(Mono.error(new AmqpException(false, "some error occurred.", null)));
+
+        // Act & Assert
+        StepVerifier.create(sessionReceiver.renewSessionLock(SESSION_ID))
+            .verifyErrorMatches(throwable -> {
+                Assertions.assertTrue(throwable instanceof ServiceBusAmqpException);
+                final ServiceBusErrorSource actual = ((ServiceBusAmqpException) throwable).getErrorSource();
+                Assertions.assertEquals(ServiceBusErrorSource.RENEW_LOCK, actual);
+                return true;
+            });
+    }
+
+    /**
+     * Verifies that error source is populated when there is any error during message settlement.
+     */
+    @ParameterizedTest
+    @MethodSource
+    void errorSourceOnSettlement(DispositionStatus dispositionStatus, ServiceBusErrorSource expectedErrorSource,
+        DeliveryStateType expectedDeliveryState) {
+        final String lockToken1 = UUID.randomUUID().toString();
+
+        final OffsetDateTime expiration = OffsetDateTime.now().plus(Duration.ofMinutes(5));
+
+        final MessageWithLockToken message = mock(MessageWithLockToken.class);
+
+        when(messageSerializer.deserialize(message, ServiceBusReceivedMessage.class)).thenReturn(receivedMessage);
+
+        when(receivedMessage.getLockToken()).thenReturn(lockToken1);
+        when(receivedMessage.getLockedUntil()).thenReturn(expiration);
+
+        when(amqpReceiveLink.updateDisposition(eq(lockToken1), argThat(e -> e.getType() == expectedDeliveryState)))
+            .thenReturn(Mono.error(new AmqpException(false, "some error occurred.", null)));
+
+        // Act & Assert
+        StepVerifier.create(receiver.receiveMessages().take(1)
+            .flatMap(context -> {
+                final Mono<Void> operation;
+                switch (dispositionStatus) {
+                    case DEFERRED:
+                        operation = receiver.defer(receivedMessage);
+                        break;
+                    case ABANDONED:
+                        operation = receiver.abandon(receivedMessage);
+                        break;
+                    case COMPLETED:
+                        operation = receiver.complete(receivedMessage);
+                        break;
+                    case SUSPENDED:
+                        operation = receiver.deadLetter(receivedMessage);
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unrecognized operation: " + dispositionStatus);
+                }
+                return operation;
+            }))
+            .then(() -> messageSink.next(message))
+            .expectNext()
+            .verifyErrorMatches(throwable -> {
```

Comment:
These should all be: verifyErrorSatisfies.  verifyErrorMatches should be taking a predicate. This predicate ALWAYS returns true because you've done assertions. So this should be a consumer. 
https://projectreactor.io/docs/test/release/api/reactor/test/StepVerifier.LastStep.html#verifyErrorSatisfies-java.util.function.Consumer-